src/Result.php

```diff
@@ -35,11 +41,17 @@ public static function invalid(array $errors): ResultInterface
         return new self(false, $errors);
     }
 
+    /**
+     * @return bool
+     */
     public function isValid(): bool
     {
         return $this->valid;
     }
 
+    /**
+     * @return array<ErrorInterface>
```

Comment:
In my opinion this one of the few examples where an additional doc comment is helpful.